### PR TITLE
MRG+1: reading raw if channels are not reordered

### PR DESCRIPTION
**Pull request by dengemann**
cc @Eric89GXL this fixes a bug that's been existing since you added the preloading functionality.
And another test.

### CODE REVIEW

File: mne/io/bti/bti.py [hunk 1]
```diff
@@ -953,7 +953,7 @@ def _read_bti_header(pdf_fname, config_fname, sort_by_ch_name=True):
         info['order'] = sort_by_name_idx
     else:
         info['chs'] = chans
-        info['order'] = Ellipsis
+        info['order'] = None
```
Comment (Eric89GXL) [inline mne/io/bti/bti.py]: if you make this np.arange(len(chans))` then you don't need the conditional later, right?

File: mne/io/bti/bti.py [hunk 2]
```diff
@@ -1063,7 +1063,13 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
...
                 data_view = data[:, sample_start:sample_stop]
                 one = np.empty(block.shape[::-1])
-                for ii, b_i_o in enumerate(bti_info['order']):
+
+                order = bti_info['order']
+                if order is None:
+                    order_ = np.arange(block.shape[1])
+                else:
+                    order_ = bti_info['order']
+                for ii, b_i_o in enumerate(order_):
```
Comment (Eric89GXL) [inline mne/io/bti/bti.py]: then these 6 lines can just be `for ii, b_i_o in enumerate(bti_info['order'])`

Comment (agramfort) [inline mne/io/bti/bti.py]: but then you cannot test easily that the order what present or not in the data

File: mne/io/bti/tests/test_bti.py [hunk 3]
```diff
@@ -195,6 +195,10 @@ def test_info_no_rename_no_reorder_no_pdf():
             np.array([ch['loc'] for ch in info['chs']]),
             np.array([ch['loc'] for ch in info2['chs']]))
 
+        # just check reading data | corner case
+        read_raw_bti(pdf_fname=pdf, config_fname=config, head_shape_fname=None,
+                     sort_by_ch_name=False, preload=True)
```
Comment (agramfort) [inline mne/io/bti/tests/test_bti.py]: you're not really testing anything right?

Comment (dengemann) [inline mne/io/bti/tests/test_bti.py]: It's a non-crash test. The bug was about preloading the data. Maybe we should test the data ordering too.

Comment (agramfort) [inline mne/io/bti/tests/test_bti.py]: if it's possible please do.

File: mne/io/bti/bti.py (hunk 2 above)
Comment (Eric89GXL) [inline mne/io/bti/bti.py]: Do we actually need to test that?

Comment (agramfort) [inline mne/io/bti/bti.py]: I was just pointing that what you suggest is a bit different in behavior :)

but whatever @dengemann agrees on is fine with me.

**Comment by dengemann** [comment]
Alright @agramfort @Eric89GXL addressed our discussion. I'm happy here. Merge if you are happy too.

**Comment by Eric89GXL** [comment]
LGTM +1 for merge

**Comment by agramfort** [comment]
+1 for merge when CIs are happy

**Comment by Eric89GXL** [comment]
Thanks @dengemann
